core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -406,7 +407,15 @@ public void run() {
       }
     }
 
-    scheduleConnectionTask = helper.getSynchronizationContext().schedule(
+    SynchronizationContext synchronizationContext = null;
+    try {
+      synchronizationContext = helper.getSynchronizationContext();
+    } catch (NullPointerException e) {
+      // If we have a test helper or old helper without a context, we can't schedule a task
```

Comment:
Can we fix the test instead of the code?

Reply:
I did fix the tests to provide a synchronization context, but customers could be relying on an old helper that they don't expect to suddenly start failing, so we need to handle the case.

Reply:
The test I can believe it can be not implemented.  But don't mention "test helper" in code comment.
The LB helper should always have `getSynchronizationContext()`, from the customer. Originally it is from ManagedChannelImpl. Where is the old helper that does not have this?

Reply:
I've updated the comment.  Despite not having any known examples that are not providing the sync context I don't want to break something unknown out in the wild.